linera-core/src/chain_worker/cache.rs

```diff
@@ -0,0 +1,259 @@
+// Copyright (c) Zefchain Labs, Inc.
+// SPDX-License-Identifier: Apache-2.0
+
+//! Types to handle a set of running chain workers.
+//!
+//! The [`ChainWorkers`] type contains a set of running [`ChainWorkerActor`]s. It will
+//! limit itself to a maximum number of active tasks, and will evict the least-recently
+//! used chain worker when it is full and a new chain worker actor needs to be created.
+
+use std::{
+    collections::{HashMap, HashSet},
+    num::NonZeroUsize,
+    sync::{Arc, Mutex, RwLock},
+};
+
+use linera_base::{crypto::CryptoHash, identifiers::ChainId};
+use linera_chain::{data_types::ExecutedBlock, types::Hashed};
+use linera_storage::Storage;
+use lru::LruCache;
+use tokio::sync::{mpsc, OwnedSemaphorePermit, Semaphore};
```

Comment:
Sometimes we use `tokio::sync::Semaphore` and sometimes `async_lock::Semaphore` - do we always make a decision which one to choose based on some specific requirements? if so, can you share?

Reply:
Ah, I didn't think of that. Thanks!.